web/src/inbox_ui.ts

```diff
@@ -946,24 +946,22 @@ export function get_focused_row_message(): {message?: Message | undefined} & (
         return {message};
     }
 
-    const $stream = $focused_row.parent(".inbox-topic-container").parent();
-    const stream_key = $stream.attr("id");
-    assert(stream_key !== undefined);
-    const row_info = topics_dict.get(stream_key)!.get(conversation_key);
-    assert(row_info !== undefined);
-    const message = message_store.get(row_info.latest_msg_id);
+    // Last case: focused on a topic row.
     // Since inbox is populated based on unread data which is part
     // of /register request, it is possible that we don't have the
     // actual message in our message_store. In that case, we return
     // a fake message object.
-    if (message === undefined) {
-        return {
-            msg_type: "stream",
-            stream_id: row_info.stream_id,
-            topic: row_info.topic_name,
-        };
-    }
-    return {message};
+    const $topic_menu_elt = $focused_row.find(".inbox-topic-menu");
+    const topic = $topic_menu_elt.attr("data-topic-name");
+    assert(topic !== undefined);
+    const stream_id = Number($topic_menu_elt.attr("data-stream-id"));
+    assert(stream_id !== undefined);
+
+    return {
+        msg_type: "stream",
+        stream_id,
+        topic,
+    };
```

Comment:
Probably we can rename `get_focused_row_message` to `get_focused_row_details` -- since it isn't ever a message object.

Reply:
We are still returning `message` for DM rows. 

```
    if (is_dm) {
        const row_info = dms_dict.get(conversation_key);
        assert(row_info !== undefined);
        const message = message_store.get(row_info.latest_msg_id);
        if (message === undefined) {
            const recipients = people.user_ids_string_to_emails_string(row_info.user_ids_string);
            assert(recipients !== undefined);
            return {
                msg_type: "private",
                private_message_recipient: recipients,
            };
        }
        return {message};
    }
    ```